pyomo/util/calc_var_value.py

```diff
@@ -81,7 +81,7 @@ def calculate_variable_from_constraint(
 
     """
     # Leverage all the Constraint logic to process the incoming tuple/expression
-    if not isinstance(constraint, _ConstraintData):
+    if not getattr(constraint, 'ctype', None) is Constraint:
         constraint = Constraint(expr=constraint, name=type(constraint).__name__)
         constraint.construct()
```

Comment:
Good catch!  You are correct: passing an IndexedConstraint to this used to get passed to the Constraint() constructor where it would die with:
```
KeyError: "Cannot treat the scalar component 'ScalarConstraint' as an indexed component"
```
Now, the indexed constraint gets to line 88 and dies with
```
AttributeError: 'IndexedConstraint' object has no attribute 'body'
```
...neither of these are great (it is neither an informative error message, nor the "correct" exception type).  We should probably add a check for `isindexed()` and raise a ValueError.  I'll push that momentarily.